src/pybind/mgr/dashboard/services/sso.py

```diff
@@ -137,30 +137,30 @@ def _get_optional_attr(cmd, attr, default):
 
 
 def handle_sso_command(cmd):
+    ret = -errno.ENOSYS, '', ''
     if cmd['prefix'] not in ['dashboard sso enable saml2',
                              'dashboard sso disable',
                              'dashboard sso status',
                              'dashboard sso show saml2',
                              'dashboard sso setup saml2']:
         return -errno.ENOSYS, '', ''
 
+    if not python_saml_imported:
+        return -errno.EPERM, '', 'Required library not found: `python3-saml`'
+
```

Comment:
A deeper refactoring on this area would've been using the `@CLICommand` helper with separate functions, instead of dealing with each command handling (the old way). If you're keen on doing that I can point you to some places where Dashboard is now using the easy approach.

Reply:
I agree but, I think this should be addressed in another diff. I don't mind doing it tho if I get some spare time from the cache.